src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -80,12 +81,35 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
 
     String user = userService.getCurrentUser().getEmail();
-    String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
-    float sentimentScore = getSentimentScore(text);
-    
-    Message message = new Message(user, text, sentimentScore);
-    datastore.storeMessage(message);
 
+    String userText = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+
+    String regex = "(https?://\\S+\\.(png|jpg|gif))";
+
+    String replacement = "<img src=\"$1\" />";
+
+
+    String youtube_regex = "(https://www.youtube.com/watch\\?v=(\\S*))";
+    String youtube_replacement = "<iframe width=\"560\" height=\"315\" "+
+   "src=\"https://www.youtube.com/embed/$2\" frameborder=\"0\" "+
+   "allow=\"accelerometer; autoplay; encrypted-media; gyroscope; "+
+   "picture-in-picture\" allowfullscreen></iframe>";
+
+    String result = userText.replaceAll(youtube_regex, youtube_replacement);
+    String textWithImagesReplaced = userText.replaceAll(regex, replacement);
+    float sentimentScore;
+
+    if(userText.equals(result)){
```

Comment:
Why is this check using just the YouTube result?

Reply:
if i dont include the check then two messages will be outputted. One message with the picture and the second message without the picture just the URL. So i included the check to compare the usertext. The if statement basically checks if the usertext has been replaced. If the usertext has been replaced then it'll continue. If that makes sense @dj-smith

Reply:
I see now what you mean, it basically relates to my other comment about how the code's repeated. The if case is using the image text, and the else case is using the YT text.